storage/bootstrap/bootstrapper/peers/source.go

```diff
@@ -677,3 +778,54 @@ func (s *peersSource) markIndexResultErrorAsUnfulfilled(
 	}
 	r.Add(result.IndexBlock{}, unfulfilled)
 }
+
+func initialTopologyState(opts Options) (topologyState, error) {
+	session, err := opts.AdminClient().DefaultAdminSession()
+	if err != nil {
+		return topologyState{}, err
+	}
+
+	topology, err := session.Topology()
+	if err != nil {
+		return topologyState{}, err
+	}
+
+	var (
+		topoMap       = topology.Get()
+		hostShardSets = topoMap.HostShardSets()
+		topologyState = topologyState{
+			majorityReplicas: topoMap.MajorityReplicas(),
+			shardStates:      shardStates{},
+		}
+	)
+
+	for _, hostShardSet := range hostShardSets {
+		for _, currShard := range hostShardSet.ShardSet().All() {
+			shardID := currShard.ID()
+			existing, ok := topologyState.shardStates[shardID]
+			if !ok {
+				existing = map[string]hostShardState{}
+				topologyState.shardStates[shardID] = existing
+			}
+
+			existing[hostShardSet.Host().String()] = hostShardState{
+				host:       hostShardSet.Host(),
+				shardState: currShard.State(),
+			}
+		}
+	}
+
+	return topologyState, nil
+}
+
+type topologyState struct {
+	majorityReplicas int
+	shardStates      shardStates
+}
+
+type shardStates map[uint32]map[string]hostShardState
```

Comment:
very nit: Maybe to make this slightly more readable you could `type hostID string` and `type shardID uint32` so it becomes `map[shardID]map[hostID]hostShardState`?

Reply:
sure